sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/WhereQuery.cs

```diff
@@ -24,15 +24,30 @@ internal WhereQuery(AdtQueryBuilder parent)
         }
 
         /// <summary>
-        /// Adds a WHERE and its conditional argument(s) clause to the query object. Meant to be used for simple
-        /// conditions involving operators or with basic ADT functions. Multiple WHERE clauses are appended using
-        /// the AND logical operator.
+        /// Adds a WHERE and the conditional arguments for a comparison to the query object. Used to compare ADT properties
+        /// using the query language's comparison operators.
         /// </summary>
-        /// <param name="condition"> A custom object that encodes the logical statement nested within the WHERE clause. </param>
+        /// <param name="field"> The field being checked against a certain value. </param>
+        /// <param name="comparisonOperator"> The comparison operator being invoked. </param>
+        /// <param name="value"> The value being checked against a Field. </param>
         /// <returns> ADT query that already contains SELECT and FROM. </returns>
-        internal WhereQuery Where(ConditionBase condition)
+        public WhereQuery WhereComparison(string field, QueryComparisonOperator comparisonOperator, string value)
```

Comment:
As I commented earlier, I think all these APIs can be named `Where` or if you decide to go down the path of adding a `Where()` at the beginning and following it up with all the conditions, you can use `IsTrue` ... `WhereComparison` is not a great method name IMHO

Reply:
same goes for `WhereContains`, `WhereOverride`

Reply:
I've changed `WhereComparison` and `WhereContains`, but `WhereOverride` matches with `SelectOverride` and `FromOverride` so I think it makes sense to keep that one as is, unless we should rename those as well. Lmk what you think about that

Reply:
Nevermind, saw your other comments and changed them to all be overloaded.